File: src/register_file.zig
const RegisterFile = struct {
    general_purpose_registers: u8[0x10],
    index_register: u16,
    program_counter: u16,
    stack_pointer: u16,
    delay_timer_register: u8,
    sound_timer_register: u8,
    current_operand: u16,
};

File: cpu/registerfile.go
package cpu

import "github.com/AlanRostem/mu-8/mu8"

const GeneralPurposeRegisterCount = 0x10

type RegisterFile struct {
	GeneralPurpose [GeneralPurposeRegisterCount]mu8.Byte
	Index          mu8.Word
	ProgramCounter mu8.Word
	StackPointer   mu8.Word
	DelayTimer     mu8.Byte
	SoundTime      mu8.Byte
	CurrentOperand mu8.Word
}

func NewRegisterFile() *RegisterFile {
	return &RegisterFile{
		GeneralPurpose: [GeneralPurposeRegisterCount]mu8.Byte{},
	}
}
